knimin/lib/data_access.py

```diff
@@ -3800,6 +3802,162 @@ def delete_pool(self, pool_id):
             TRN.add(sql, [pool_id])
             TRN.execute()
 
+    def condense_dna_plates(self, dna_plates, name, email, robot,
+                            plate_type, volume):
+        """Generates a new shotgun plate
+
+        Parameters
+        ----------
+        dna_plates : list of [(int, int)]
+            The dna plate and the position in which they are condensed
+        name : str
+            The plate name
+        email : str
+            The email of the user
+        robot : str
+            The robot performing the plate condensation
+        plate_type : int
+            The plate type id of the shotgun plate
+        volume : float
+            The volume plated in the shotgun plate from the original DNA plates
+
+        Returns
+        -------
+        int
+            The id of the new shotgun plate
+
+        Raises
+        ------
+        ValueError
+            If plate_type is not a 384-well plate and dna_plates are not four
+            96-well plates in positions from 0-3.
+
+        Notes
+        -----
+        Right know we only know one way of condensing plates, as other
+        configurations are known, they can be added. This is coded like this to
+        make the system more flexible
+        """
+        with TRN:
+            self._sample_plate_is_unique(name)
+            self._plate_type_exists(plate_type)
+            self._email_exists(email)
+            robot_id = self.get_or_create_property_option_id(
+                "extraction_robot", robot)
+
+            # 2 is 384-well plate
+            if plate_type != 2:
+                raise ValueError("plate_map should be 2: '384-well plate' but "
+                                 "it's %d" % plate_type)
+            if len(dna_plates) != 4:
```

Comment:
Can you modify to check if 0 < len(dna_plates) <= 4 (and fix error message below accordingly?